File: java/src/main/java/problems/Problem051TransposeMatrix.java
package problems;


public class Problem051TransposeMatrix {
    public static int[][] transpose(int[][] M){
        int m=M.length, n=M[0].length;
        int[][] T=new int[n][m];
        for(int i=0;i<m;i++) for(int j=0;j<n;j++) T[j][i]=M[i][j];
        return T;
    }
}

File: scala/src/main/scala/problems/Problem051TransposeMatrix.scala

package problems

object Problem051TransposeMatrix {
  def transpose(M:Array[Array[Int]]): Array[Array[Int]] = {
    val m=M.length; val n=M(0).length
    val T = Array.ofDim[Int](n,m)
    var i=0; while (i<m) { var j=0; while (j<n) { T(j)(i)=M(i)(j); j+=1 }; i+=1 }
    T
  }
}

